src/service.rs

```diff
@@ -197,8 +213,43 @@ pub struct Service {
     /// A queue of peers that require regular ping to check connectivity.
     peers_to_ping: HashSetDelay<NodeId>,
 
+    /// A receiver/initiator peer that a hole-punch PING is sent to is stored to make sure only
+    /// one relay is used at a time when trying to hole-punch a NAT and to keep track of
+    /// connection direction in session establishment.
+    hole_punch_pings: HashSet<NodeId>,
+
     /// A channel that the service emits events on.
     event_stream: Option<mpsc::Sender<Discv5Event>>,
+
+    /// If this Discv5 instance is configured to allow peers behind symmetric NATs
```

Comment:
Why is this stored by distance? I couldn't find a strong use case for this

Reply:
if this is needed, it would be better to use the Log2Distance alias to make it clear

Reply:
To check that we don't have more than `nat_symmetric_limit` peers behind a symmetric NAT per kbucket without intreating through all peers in a kbucket and checking their ENRs.